hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -401,9 +401,9 @@ protected Object putInternal(final K key, final V value, final ExpiryPolicy expi
             }
         } else {
             if (nearCache != null || statisticsEnabled) {
-                future.andThen(new ExecutionCallback<Object>() {
```

Comment:
Yes, adds conplexity but just for cache async put. And you know this request comes from customer. If they are ok without this, I am OK